tests/pulse/test_circuit_to_pulses.py

```diff
@@ -463,11 +463,7 @@ def test_translate(self, platform):  # pylint: disable=R0914 # disable pyling to
 
         pulse_schedule = pulse_schedules[0]
         # there are 6 different buses + 3 empty for unused flux lines
-        assert len(pulse_schedule) == 9
-        assert all(len(schedule_element.timeline) == 0 for schedule_element in pulse_schedule.elements[-3:])
-
-        # we can ignore empty elements from here on
-        pulse_schedule.elements = pulse_schedule.elements[:-3]
+        assert len(pulse_schedule) == 6
```

Comment:
In this test we don't know if the flux buses contains AWGs or not.

I would like to have dedicated tests for this, where you modify the instruments inside a bus by adding/removing an AWG, and you check explicitly that the empty pulse schedules are added or not

Reply:
fair enough, including those tests.

Reply:
unittest added to check that adding/removing buses work as expected, please review that the buses added with the circuit in the unittest makes sense in the number of pulse schedules created.